lib/data/post_repository.dart

```diff
@@ -0,0 +1,28 @@
+import 'dart:convert';
+
+import 'package:chopper/chopper.dart' hide Post;
+import 'package:vasili_mentoring_tasks/data/models/post_remote.dart';
+import 'package:vasili_mentoring_tasks/data/posts_api_service.dart';
+import 'package:vasili_mentoring_tasks/domain/models/post.dart';
+
+class PostRepository {
+  final PostApiService apiService;
+
+  PostRepository({this.apiService});
+
+  Future<List<Post>> getAll() async {
+    Response response = await apiService.getPosts();
+    if (!response.isSuccessful) return Future.error("Error getting posts");
+    List<dynamic> list = jsonDecode(response.bodyString);
+    List<PostRemote> remotePosts = list.map((jsonObject) => PostRemote.fromJson(jsonObject)).toList();
+    return remotePosts.map((remotePost) => remotePost.toDomainModel()).toList();
+  }
+
+  Future<Post> getOne(int id) async {
+    Response response = await apiService.getPost(id);
+    if (!response.isSuccessful) return Future.error("Error getting post $id");
+    Map<String, dynamic> jsonObject = jsonDecode(response.bodyString);
+    PostRemote postRemote = PostRemote.fromJson(jsonObject);
```

Comment:
в дарте по конвеншену не обязательно и даже рекомендовано не указывать типы слева.
И опять же, final as default

Reply:
Good point, I have changed it for the other cases but here I prefer to leave types because it hard to understand which types are returns from the right-side expressions and explicitly specifying them helps when reading.